mdtraj/utils/unit/mymatrix.py

```diff
@@ -235,7 +235,7 @@ def __mul__(self, rhs):
             for j in range(n):
                 for k in range(r):
                     result[i][j] += self[i][k] * rhs[k][j]
-        return result
+        return np.float64(result)
```

Comment:
I have so many thoughts about this module. 

Does it make sense to cast the result as np.float64 when the calculations weren't necessarily done with `np.float64` in the first place?

More broadly, does it make sense to have a **pure Python** linear algebra matrix when NumPy is already a dependency? Should we be coverting to and working with NumPy arrays here instead of looping over lists?

Reply:
I don't see numpy imported in this module. Is it coming from somewhere else?

Reply:
I must've missed it. :/

Reply:
Adding to the (my) confusion, the entire units module is vendored from OpenMM and ... well I'm not sure what it does here

Reply:
> Adding to the (my) confusion, the entire units module is vendored from OpenMM and ... well I'm not sure what it does here

It does all sorts of unit conversions. I did a quick look and it seems like not much have changed. Just looking ahead, I'm assuming all our changes here could be backported to openmm if/when they decide to support numpy 2.